src/mlpack/methods/time_series_forecast/exponential_smoothing/ses.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file methods/time_series/exponential_smoothing/ses.cpp
+ * @author Aditi Pandey
+ *
+ * Implementation of Single exponential smoothing.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/core/util/log.hpp>
+#include "ses.hpp"
+#include <cmath>
+#include <numeric>
+
+using namespace mlpack;
+using namespace mlpack::ts;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+// Calculate errors
+
+double CalcMae(arma::rowvec& errors) {
+    double mae = 0.0;
+    for(arma::uword c=0; c < errors.n_elem; ++c){
+	mae += std::abs(errors(c));
+    }
+    return mae/errors.n_elem;
+}
+
+double CalcMse(arma::rowvec& errors) {
+    double mse = 0.0;
+    for(arma::uword c=0; c < errors.n_elem; ++c){
+	mse += pow(errors(c), 2);
+    }
+    return mse/errors.n_elem;
+}
+
+double CalcMape(arma::rowvec& data, arma::rowvec& errors) {
+    double mape = 0.0;
+    for(arma::uword c1=0, c2=0; c1 < errors.n_elem, c2 < data.n_elem; ++c1, ++c2){
+	    mape += std::abs(errors(c1) / data(c2));
+    }
+    return mape/errors.n_elem;
+}
+
+//Function declaration for the SES 
+
+SingleES::SingleES(arma::rowvec& data, const double& alpha) {
+    this->alpha = alpha;
+    int sZ =0;
+    // forecastPast.push_back(data.front());
+    forecastPast = {data.front()};
+    
+    //  errorForecastPast.push_back(0);
+    errorForecastPast = {0};
+
+    // Evaluate the output with the help of smoothing factor
+    
+    for(arma::uword c=0; c < data.n_elem-1; ++c){
```

Comment:
Have you seen the mlpack style guide?  https://github.com/mlpack/mlpack/wiki/DesignGuidelines#style-guidelines  A lot of stuff here should be adapted to that before we merge.  If you can take a full pass over all the code, I'd appreciate it. :)

Reply:
Yes surely, I thought I would do the code cleaning in the end, but sure I'll do then in the next update.